server/controllers/chartsController/index.js

```diff
@@ -1,80 +1,156 @@
-import { ObjectID } from 'mongodb'
-
+import { ObjectId } from 'mongodb'
+import deepEqual from 'deep-equal'
 import { collections } from '../../utils/mongoCollections'
 import { SITE_NAMES } from '../../utils/siteNames'
 import {
   generateStudyTargetTotals,
   isAnyTargetIncluded,
-  N_A,
   postProcessData,
   processData,
   processTotals,
-  TOTALS_STUDY,
+  mongoQueryFromFilters,
+  sanitizeQueryParams,
 } from './helpers'
+import {
+  N_A,
+  TOTALS_STUDY,
+  STUDIES_TO_OMIT,
+  DEFAULT_CHART_FILTERS,
+  EMPTY_VALUE,
+  INCLUSION_EXCLUSION_CRITERIA_FORM,
+} from '../../constants'
 
-const STUDIES_TO_OMIT = ['files', 'combined']
-
-export const graphDataController = async (dataDb, userAccess, chart_id) => {
+export const graphDataController = async (
+  dataDb,
+  userAccess,
+  chart_id,
+  parsedQueryParams
+) => {
   const labelMap = new Map()
   const dataMap = new Map()
+  const areFiltersRequested =
+    !!Object.keys(parsedQueryParams).length &&
+    !deepEqual(DEFAULT_CHART_FILTERS, sanitizeQueryParams(parsedQueryParams))
+  const filters = areFiltersRequested
+    ? sanitizeQueryParams(parsedQueryParams)
+    : DEFAULT_CHART_FILTERS
   const chart = await dataDb
     .collection(collections.charts)
-    .findOne({ _id: ObjectID(chart_id) })
-  const allSubjects = await dataDb
-    .collection(collections.toc)
-    .find(
-      {
-        assessment: chart.assessment,
-        study: { $in: userAccess, $nin: STUDIES_TO_OMIT },
-      },
-      { projection: { collection: 1, study: 1, _id: 0 } }
-    )
-    .toArray()
-
+    .findOne({ _id: ObjectId(chart_id) })
+  const allSubjects = await getAllSubjects({
+    dataDb,
+    chart,
+    userAccess,
+    areFiltersRequested,
```

Comment:
do we need to pass this in to the function? What if `filters` was set to `undefined` if there are no filters or if all filters are selected? Then we can determine if we need to filter the subjects based on that value of `filters`.

Reply:
We need it or something similar so could use your thoughts for this scenario.
When all the filters are selected we want the default behavior, but if there are subjects that have not taken the form_inclusionexclusion_criteria_review assessment, then we would be missing those folks from the counts.

The boolean here, has a deep equality check that if all filters are selected then we want the default behavior path.

What is your perspective of this scenario?

Reply:
I think it will be simpler to just pass the filters parameter and then use the default behavior if `filters` is `undefined` or if `filters` equals `DEFAULT_CHART_FILTERS`.